src/download.js

```diff
@@ -1,81 +1,190 @@
+ * @param {string} server one of SERVER_BIORXIV or SERVER_MEDRXIV
+ * @param {number|string} N the number of posts OR days of posts (number followed by letter 'd')
+ * @param {object} [opts] fetch opts
+ * @returns { array } normalized metadata: If N is number, max items is 100, else not sure
+ */
+async function getRecent (server, N, opts) { // eslint-disable-line no-unused-vars
+  validateDay(N);
+  return getNormalized(`${CONTENT_DETAIL_BASE_URL}/${server}/${N}`, opts);
+};
+
+/**
+ * getDateRange
+ *
+ * Get metadata for articles between the specified dates
+ * NB: 'Where metadata for multiple papers is returned, results are paginated with 100 papers served in a call.'
+ *   - see {@link http://api.biorxiv.org/}
+ * @param {string} server one of SERVER_BIORXIV or SERVER_MEDRXIV
+ * @param {string} start date (yyyy-mm-dd)
+ * @param {string} end date (yyyy-mm-dd)
+ * @param {number} [offset] cursor-like index to start return from (default 0)
+ * @param {object} [opts] fetch opts
+ * @returns { object } normalized metadata
+ */
+async function getDateRange (server, start, end, offset = 0, opts) {
+  validateDate(start);
+  validateDate(end);
+  const url = `${CONTENT_DETAIL_BASE_URL}/${server}/${start}/${end}/${offset}/json`;
+  return getNormalized(url, opts);
+};
+
+// ********* Public API *********** //
+
+/**
+ * download
+ *
+ * Get metadata for articles between the specified dates
+ * @param {string} server one of SERVER_BIORXIV or SERVER_MEDRXIV
+ * @param {string} start date (yyyy-mm-dd)
+ * @param {string} end date (yyyy-mm-dd)
+ * @returns {array} list of normalized article metadata items
+ */
+export async function download (server, start, end) {
+  validateServer(server);
+  const FETCH_OPTS = {
+    method: 'GET',
+    headers: {
+      Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+      'Accept-Encoding': 'gzip, deflate, br',
+      'Accept-Language': 'en-US,en;q=0.9',
+      'Cache-Control': 'no-cache',
+      Connection: 'keep-alive',
+      Host: 'api.biorxiv.org',
+      Pragma: 'no-cache',
+      'sec-ch-ua': '"Not A;Brand";v="99", "Chromium";v="101", "Google Chrome";v="101"',
+      'sec-ch-ua-mobile': '?0',
+      'sec-ch-ua-platform': '"macOS"',
+      'Sec-Fetch-Dest': 'document',
+      'Sec-Fetch-Site': 'none',
+      'Sec-Fetch-User': '?1',
+      'Upgrade-Insecure-Requests': '1',
+      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36'
+    }
+  };
+
+  try {
+    let N = 0;
+    let offset = 0;
+    let items = [];
+
+    // use cursor to iterate over total
+    do {
+      const { count, total, collection } = await getDateRange(server, start, end, offset, FETCH_OPTS);
+      N = total;
+      offset += count;
+      items = items.concat(collection);
+    } while (offset < N);
+
+    return items;
+  } catch (err) {
```

Comment:
Not high-priority, but good to know:  The try-catch could be within the loop so that we could retry if just a subquery fails.

Reply:
To be safe, I would set a retry counter so there's no endless loop.

Reply:
Yeah, I don't think it's important right now, but it was worth mentioning.  It's worth revisiting once we put up a cron job.  We'll have to handle failure and retrying at some level there.  We don't want the cron job to fail completely so that we have missing data.  

We'll have to see how frequent errors are in practice with a running cron job and re-evaluate.